pkg/controller/backup_controller.go

```diff
@@ -307,7 +320,10 @@ func (b *backupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	log.Info("Updating backup's final status")
 	if err := kubeutil.PatchResource(original, request.Backup, b.kbClient); err != nil {
 		log.WithError(err).Error("error updating backup's final status")
+		// return the error so the status can be re-processed; it's currently still not completed or failed
```

Comment:
I remember the CR won't be re-enqueued when returning an error from the reconciler, could you confirm this?

Reply:
We had confirmed that returning error would cause reconciler to requeue.

This is equivalent to returning with `requeue: true` but with more information.

See: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.4/pkg/reconcile#Reconciler

Reply:
```
	// If the returned error is non-nil, the Result is ignored and the request will be
	// requeued using exponential backoff. The only exception is if the error is a
	// TerminalError in which case no requeuing happens.
```